pkg/controller/windowsmachineconfig/windows/windows.go

```diff
@@ -90,6 +103,95 @@ func (vm *Windows) initializeBootstrapperFiles() error {
 	return nil
 }
 
+// handleHybridOverlay ensures that the hybrid overlay is running on the node
+func (vm *Windows) handleHybridOverlay(nodeName string) error {
+	// Check if the hybrid-overlay is running
+	_, stderr, err := vm.Run("Get-Process -Name \""+hybridOverlayProcess+"\"", true)
+
+	// stderr being empty implies that hybrid-overlay was running.
+	if err == nil || stderr == "" {
+		// Stop the hybrid-overlay
+		stopCmd := "Stop-Process -Name \"" + hybridOverlayProcess + "\""
+		_, stderr, err := vm.Run(stopCmd, true)
+		if err != nil || stderr != "" {
+			log.Info("unable to stop hybrid-overlay", "stop command", stopCmd, "stderr", stderr)
+			return errors.Wrap(err, "unable to stop hybrid-overlay")
+		}
+	}
+
+	_, stderr, err = vm.Run(mkdirCmd(logDir), false)
+	if err != nil {
+		return errors.Wrap(err, fmt.Sprintf("unable to create %s directory:\n%s", logDir, stderr))
+	}
+
+	if err := vm.CopyFile(wkl.HybridOverlayPath, remoteDir); err != nil {
+		return errors.Wrap(err, fmt.Sprintf("error copying %s-->%s", wkl.HybridOverlayPath,
+			remoteDir+wkl.HybridOverlayName))
+	}
+
+	// Start the hybrid-overlay in the background over ssh. We cannot use vm.Run() and by extension WinRM.Run() here as
+	// we observed WinRM.Run() returning before the commands completes execution. The reason for that is unclear and
+	// requires further investigation.
+	// TODO: This will be removed in https://issues.redhat.com/browse/WINC-353
+	go vm.RunOverSSH(remoteDir+wkl.HybridOverlayName+" --node "+nodeName+
+		" --k8s-kubeconfig c:\\k\\kubeconfig > "+logDir+"hybrid-overlay.log 2>&1", false)
+
+	if err = vm.waitForHybridOverlayToRun(); err != nil {
+		return errors.Wrap(err, fmt.Sprintf("error running %s", wkl.HybridOverlayName))
+	}
+
+	if err = vm.waitForOpenShiftHNSNetworks(); err != nil {
+		return errors.Wrap(err, "error waiting for OpenShift HNS networks to be created")
+	}
+
```

Comment:
Should we also wait for the mac address annotation signifying that the hybrid overlay has done part of its job?

Reply:
Good point. I will add that too. Then I can remove the waits for the annotations in the test code but just check for them.